From d335d34dbee34f2223baa0154d3f621798d06b94 Mon Sep 17 00:00:00 2001
From: Guo Jie <guojie@loongson.cn>
Date: Fri, 8 Sep 2023 10:00:21 +0800
Subject: [PATCH 117/133] LoongArch: Enable -fsched-pressure by default at -O1
 and higher.

gcc/ChangeLog:

	* common/config/loongarch/loongarch-common.cc:
	(default_options loongarch_option_optimization_table):
	Default to -fsched-pressure.

Signed-off-by: Peng Fan <fanpeng@loongson.cn>
Signed-off-by: ticat_fp <fanpeng@loongson.cn>
---
 gcc/common/config/loongarch/loongarch-common.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/common/config/loongarch/loongarch-common.cc b/gcc/common/config/loongarch/loongarch-common.cc
index 309fcb28066..c8bc5718d47 100644
--- a/src/gcc/common/config/loongarch/loongarch-common.cc
+++ b/src/gcc/common/config/loongarch/loongarch-common.cc
@@ -36,6 +36,7 @@ static const struct default_options loongarch_option_optimization_table[] =
   { OPT_LEVELS_ALL, OPT_fasynchronous_unwind_tables, NULL, 1 },
   { OPT_LEVELS_1_PLUS, OPT_fsection_anchors, NULL, 1 },
   { OPT_LEVELS_2_PLUS, OPT_free, NULL, 1 },
+  { OPT_LEVELS_1_PLUS, OPT_fsched_pressure, NULL, 1 },
   { OPT_LEVELS_NONE, 0, NULL, 0 }
 };
 
-- 
2.43.4

